packages/kernel-6.1/latest-kernel-full-config.sh

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env bash
+
+#
+# Common error handling
+#
+
+exit_trap_cmds=()
+
+on_exit() {
+    exit_trap_cmds+=( "$1" )
+}
+
+run_exit_trap_cmds() {
+    for cmd in "${exit_trap_cmds[@]}"; do
+        eval "${cmd}"
+    done
+}
+
+trap run_exit_trap_cmds EXIT
+
+bail() {
+    if [[ $# -gt 0 ]]; then
+        >&2 echo "Error: $*"
+    fi
+    exit 1
+}
+
+# Function to display usage information
+usage() {
+    cat << EOF
+Usage: $(basename "$0") -v VERSION -r RPM_FILE -s SDK_VERSION
+
+Merge kernel configurations with the following parameters:
+    -v VERSION      Kernel version
+    -r RPM_FILE    Path to RPM file
+    -s SDK_VERSION SDK version (e.g., v0.60.0)
+    -h             Display this help message
+
+Dependencies:
+    - docker
+    - rpm2cpio
+    - cpio
+    - tar
+EOF
+}
+
+usage_error() {
+    >&2 usage
+    bail "$1"
+}
+
+check_dependencies() {
+    hash rpm2cpio || usage_error "DEPENDENCY ERROR: Please install rpm2cpio somewhere in your PATH"
+    hash cpio || usage_error "DEPENDENCY ERROR: Please install cpio somewhere in your PATH"
+    hash tar || usage_error "DEPENDENCY ERROR: Please install tar somewhere in your PATH"
+    hash docker || usage_error "DEPENDENCY ERROR: Please install docker somewhere in your PATH"
+}
+
+
+while [[ $# -gt 0 ]]; do
+    case $1 in
+        -r|--rpm-file)
+            shift; rpm_file="$1" ;;
+        *)
+            usage_error "Invalid option '$1'" ;;
+    esac
+    shift
+done
+
+# Verify all required parameters are provided
+if [ -z "$rpm_file" ]; then
+    echo "Error: Missing required parameters"
+    usage
+    exit 1
+fi
+
+# expect $pwd to be packages/kernel-${kver}
+merge_kernel_configs() {
```

Comment:
If I interrupt this script, I'm left with a dirty tree. E.g.:

```
On branch develop

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	config-aarch64
	config-full-bottlerocket-aarch64
	config-x86_64
	kernel.spec
	latest-kernel-full-config.sh
	linux-6.1.131/
```

Can we either trap interrupts with a cleanup command, or run this script entirely within the SDK container?